src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -1145,6 +1145,42 @@ export class TerminalInstance implements ITerminalInstance {
 		this._xterm._core.debug = !this._xterm._core.debug;
 		this._xterm.setOption('debug', this._xterm._core.debug);
 	}
+
+	public get initialCwd(): string {
+		return this._processManager.initialCwd;
+	}
+
+	public getCwd(configHelper: ITerminalConfigHelper): Promise<string> {
```

Comment:
Sorry I wasn't clear here, the `TerminalInstance` function should only have the sourceCwd functionality (and the windows fallback). The rest can remain a helper in `terminalActions`.